src/fu-util.c

```diff
@@ -1959,16 +1965,120 @@ fu_util_update (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_modify_remote_warning (FuUtilPrivate *priv, FwupdRemote *remote, GError **error)
+{
+#if AS_CHECK_VERSION(0,7,8)
+	AsAgreement *agreement;
+	AsAgreementSection *section;
+	const gchar *tmp;
+	g_autofree gchar *basename = NULL;
+	g_autofree gchar *desc_plain = NULL;
+	g_autofree gchar *fn = NULL;
+	g_autoptr(AsApp) app = NULL;
+	g_autoptr(GHashTable) os_release = NULL;
+	g_autoptr(GString) desc_plain_str = NULL;
+	g_autoptr(SoupURI) uri = NULL;
+
+	/* nothing specified */
+	if (fwupd_remote_get_metainfo_uri (remote) == NULL)
+		return TRUE;
+
+	/* download file to usercache */
+	basename = g_path_get_basename (fwupd_remote_get_metainfo_uri (remote));
+	fn = g_build_filename (g_get_user_cache_dir (), "fwupdmgr", basename, NULL);
+	uri = soup_uri_new (fwupd_remote_get_metainfo_uri (remote));
+	if (!fu_util_download_file (priv, uri, fn, NULL, error))
+		return FALSE;
+
+	/* parse file and get default agreement section */
+	app = as_app_new ();
+	if (!as_app_parse_file (app, fn, AS_APP_PARSE_FLAG_NONE, error))
+		return FALSE;
+	agreement = as_app_get_agreement_by_id (app, "warning");
+	if (agreement == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No agreement found in AppStream file");
+		return FALSE;
+	}
+	section = as_agreement_get_section_default (agreement);
+	if (section == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No default section found in AppStream file");
+		return FALSE;
+	}
+	tmp = as_agreement_section_get_desc (section);
+	if (tmp == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No description found in agreement section");
+		return FALSE;
+	}
+
+	/* convert to console text */
+	desc_plain = as_markup_convert_simple (tmp, error);
+	if (desc_plain == NULL)
+		return FALSE;
+
+	/* replace any dynamic values from os-release */
+	desc_plain_str = g_string_new (desc_plain);
+	os_release = fwupd_get_os_release (error);
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:NAME$",
+				 g_hash_table_lookup (os_release, "NAME"));
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:BUG_REPORT_URL$",
+				 g_hash_table_lookup (os_release, "BUG_REPORT_URL"));
```

Comment:
I don't think you'll be getting this in `/etc/os-release` will you?

Reply:
I get `BUG_REPORT_URL="https://bugzilla.redhat.com/"` -- is there not something for Debian?

Reply:
Well look at that, I was expecting that since lsb_release didn't provide it it wasn't in /etc/os-release.  Glad I was wrong :)

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

$ sudo docker run -i -t debian:testing
root@3cf988413ced:/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux buster/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```